turn/src/server/mod.rs

```diff
@@ -52,43 +56,62 @@ impl Server {
             let auth_handler = Arc::clone(&s.auth_handler);
             let realm = s.realm.clone();
             let channel_bind_timeout = s.channel_bind_timeout;
-            let shutdown_rx = shutdown_rx.clone();
+            let handle_rx = handle.subscribe();
+            let conn = p.conn;
 
-            tokio::spawn(async move {
+            tokio::spawn({
```

Comment:
This whole block could just be 

```rust
let allocation_manager = Arc::new(Manager::new(ManagerConfig {
    relay_addr_generator: p.relay_addr_generator,
}));
tokio::spawn(Server::read_loop(
    conn,
    allocation_manager,
    nonces,
    auth_handler,
    realm,
    channel_bind_timeout,
    handle_rx,
));
```

No need to create an async block or do `await` since `Server::read_loop` returns a future.